### PR TITLE
Fix notification avatar

### LINKED ISSUE #1991: User avatar are not always displayed in notification
Sometimes the letter is used instead.

Also the notifications are not rendering images, as it was the case on Element Android.

### PR DESCRIPTION
**Pull request by bmarty**
## Type of change

- [ ] Feature
- [x] Bugfix
- [ ] Technical
- [ ] Other :

## Content

Use ImageLoader associated to a MatrixClient to be able to load avatars.
Create `ImageLoaderHolder` to share the `ImageHolder` between the one used by `LoggedInAppScopeFlowNode` and the one used by `DefaultNotificationDrawerManager`

## Motivation and context

Fixes #1991 

## Screenshots / GIFs

NA

## Tests

- Log in to a session
- Kill the app
- Receive a message from someone with an avatar
- Observe that now the avatar is displayed in the notification.

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s):

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-x-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR

### CODE REVIEW

File: libraries/push/impl/src/main/kotlin/io/element/android/libraries/push/impl/notifications/NotificationBitmapLoader.kt [hunk 1]
```diff
@@ -39,21 +39,22 @@ class NotificationBitmapLoader @Inject constructor(
...
 
-    private suspend fun loadRoomBitmap(path: String): Bitmap? {
+    private suspend fun loadRoomBitmap(path: String, imageLoader: ImageLoader): Bitmap? {
         return try {
             val imageRequest = ImageRequest.Builder(context)
                 .data(MediaRequestData(MediaSource(path), MediaRequestData.Kind.Thumbnail(1024)))
                 .transformations(CircleCropTransformation())
                 .build()
-            val result = context.imageLoader.execute(imageRequest)
+            val result = imageLoader.execute(imageRequest)
```
Comment (bmarty) [inline libraries/push/impl/src/main/kotlin/io/element/android/libraries/push/impl/notifications/NotificationBitmapLoader.kt:57]: That was the bug, the default `imageLoader` is not correct when there is a session, but the UI is dead (spotted by @ganfra )

File: libraries/push/impl/src/main/kotlin/io/element/android/libraries/push/impl/notifications/ImageLoaderHolder.kt
Comment (jmartinesp) [inline libraries/push/impl/src/main/kotlin/io/element/android/libraries/push/impl/notifications/ImageLoaderHolder.kt:1]: Why do we have 2 of these?

**Review by jmartinesp — COMMENTED** [review]
A question, is this expected?

File: libraries/push/impl/src/test/kotlin/io/element/android/libraries/push/impl/notifications/RoomGroupMessageCreatorTest.kt [hunk 2]
```diff
@@ -141,20 +143,7 @@ class RoomGroupMessageCreatorTest {
...
-                    coilRequests.add(it)
-                    true
-                },
-                drawable = ColorDrawable(Color.BLUE)
-            )
-            .build()
-        val imageLoader = ImageLoader.Builder(RuntimeEnvironment.getApplication())
-            .components { add(engine) }
-            .build()
-        Coil.setImageLoader(imageLoader)
```
Comment (bmarty) [inline libraries/push/impl/src/test/kotlin/io/element/android/libraries/push/impl/notifications/RoomGroupMessageCreatorTest.kt:157]: `Coil.setImageLoader(imageLoader)` is not needed anymore.

**Review by jmartinesp — APPROVED** [review]

**Pull request merged**